ledger/src/blockstore.rs

```diff
@@ -3317,6 +3317,55 @@ impl Blockstore {
         );
         Ok(())
     }
+
+    /// Mark a root `slot` as connected, traverse `slot`'s children and update
+    /// the children's connected status if appropriate.
+    ///
+    /// A ledger with a full path of blocks from genesis to the latest root will
+    /// have all of the rooted blocks marked as connected such that new blocks
+    /// could also be connected. However, starting from some root (such as from
+    /// a snapshot) is a valid way to join a cluster. For this case, mark this
+    /// root as connected such that the node that joined midway through can
+    /// have their slots considered connected.
+    pub fn set_and_chain_connected_on_root(&self, slot: Slot) -> Result<()> {
```

Comment:
nit: `set_and_chain_connected_on_root_and_children()` or `connect_child_slots_from_root()`.

Reply:
also, I'd prefer to hint the given Slot param to be expected having been _rooted_ as `, root: Slot) -> Result<...>`

Reply:
hmm, seems current codebase is inconsistent to mean the same thing: `next`, `child`, `descendant`... 

I'd prefer to stick to `next`, that's because it's set in stone (i.e. `blockstone`, lol)

Reply:
also, no new unit test for this `fn`?

Reply:
> also, I'd prefer to hint the given Slot param to be expected having been rooted as , root: Slot) -> Result<...>
Done

> I'd prefer to stick to next (taken from meta.next_slots.), that's because it's set in stone (i.e. blockstone, lol)
I'll use `next` for now to match what is currently in there, but I might make a PR to change `next_slots` to `children_slots`. I think this aligns with our naming the field `parent_slot` in `SlotMeta` as opposed to saying `previous_slot`.

> also, no new unit test for this fn?
Had it tagged as a TODO item, now that fire fighting has subsided, I can add something in.

Reply:
Unit test done